### PR TITLE
Adding test to domain mappers

### PR DESCRIPTION
**Pull request by noiarek**
**What changed?**

Models / Mappers / etc are way bigger than I thought, so I decided to add test to make sure we're storing the right values. And I found some bugs already!

1. Adding test to domain mapper 
2. Fixing incorrect type hinting (from `describe_domain` to `domain`)

**Why?**
Improving test

**How did you test it?**
Unit Test

**Potential risks**

 
**Release notes**

### CODE REVIEW

File: cadence/mapping/grpc/domain.py [hunk 1]
```diff
@@ -20,69 +26,67 @@ def list_domains_request_dataclass_to_proto(list_domains: ListDomainsRequest) ->
...
     return ListDomainsResponse(
-        domains=[proto_describe_domain_to_dataclass(domain) for domain in list_domains.domains],
+        domains=[proto_domain_to_describe_domain_response_dataclass(domain) for domain in list_domains.domains],
         next_page_token=list_domains.next_page_token
     ) if list_domains else None
 
 
-def proto_describe_domain_to_dataclass(
-        describe_domain: service_domain_pb2.DescribeDomainResponse) -> DescribeDomainResponse:
+def proto_domain_to_describe_domain_response_dataclass(domain: domain_pb2.Domain) -> DescribeDomainResponse:
```
Comment (noiarek) [inline cadence/mapping/grpc/domain.py:34]: Changing from `DescribeDomainResponse` to `Domain`

File: cadence/mapping/grpc/domain.py [hunk 2]
```diff
@@ -20,69 +26,67 @@ def list_domains_request_dataclass_to_proto(list_domains: ListDomainsRequest) ->
...
-def proto_domain_status_to_dataclass(domain_status: domain_pb2.DomainStatus) -> DomainStatus:
+def proto_domain_status_to_dataclass(domain_status: Optional[domain_pb2.DomainStatus.__class__]) -> DomainStatus:
     if domain_status == domain_pb2.DOMAIN_STATUS_REGISTERED:
         return DomainStatus(DomainStatus.REGISTERED)
     elif domain_status == domain_pb2.DOMAIN_STATUS_DELETED:
         return DomainStatus(DomainStatus.DELETED)
     elif domain_status == domain_pb2.DOMAIN_STATUS_DEPRECATED:
         return DomainStatus(DomainStatus.DEPRECATED)
-    else:
+    else: #Should this value be the default value? (it is if it's null) Does the API always return something? Maybe null?
```
Comment (noiarek) [inline cadence/mapping/grpc/domain.py:69]: Question

File: cadence/mapping/grpc/domain.py [hunk 3]
```diff
@@ -20,69 +26,67 @@ def list_domains_request_dataclass_to_proto(list_domains: ListDomainsRequest) ->
...
         return DomainStatus(DomainStatus.INVALID)
 
 
-def proto_domain_configuration_do_dataclass(
-        domain_response: service_domain_pb2.DescribeDomainResponse) -> DomainConfiguration:
+def proto_domain_to_domain_configuration_dataclass(domain: domain_pb2.Domain) -> DomainConfiguration:
     return DomainConfiguration(
-        workflow_execution_retention_period_in_days=ms_to_days(domain_response.workflow_execution_retention_period.ToMilliseconds()),
-        workflow_execution_retention_period=domain_response.workflow_execution_retention_period.ToMilliseconds(),
+        workflow_execution_retention_period_in_days=ms_to_days(domain.workflow_execution_retention_period.ToMilliseconds()),#is this milliseconds?
```
Comment (noiarek) [inline cadence/mapping/grpc/domain.py:75]: Question

File: cadence/mapping/grpc/domain.py [hunk 4]
```diff
@@ -102,7 +107,7 @@ def archival_status_dataclass_to_proto(archival_status: ArchivalStatus) -> domai
 
 def proto_bad_binaries_to_dataclass(bad_binaries: domain_pb2.BadBinaries) -> BadBinaries:
     return BadBinaries(
-        binaries={key: proto_bad_binary_info_to_dataclass(value) for key, value in bad_binaries.binaries}
+        binaries={key: proto_bad_binary_info_to_dataclass(value) for key, value in bad_binaries.binaries.items()}
```
Comment (noiarek) [inline cadence/mapping/grpc/domain.py:110]: missing `items()`

File: cadence/mapping/grpc/domain.py [hunk 5]
```diff
@@ -133,13 +139,15 @@ def register_domain_request_dataclass_to_proto(register_domain: RegisterDomainRe
...
         description=register_domain.description,
         owner_email=register_domain.owner_email,
-        workflow_execution_retention_period=duration_or_none(register_domain.workflow_execution_retention_period_in_days),
+        workflow_execution_retention_period=duration_or_none(days_to_seconds(register_domain.workflow_execution_retention_period_in_days)),
         clusters=[cluster_replication_configuration_metadata_to_proto(cluster) for cluster in register_domain.clusters],
         active_cluster_name=register_domain.active_cluster_name,
-        data={key:value for key,value in register_domain.data.items()},
+        data={key: value for key,value in register_domain.data.items()},
         is_global_domain=register_domain.is_global_domain,
+        # Default value is 0 (invalid), it doesn't accept nil
```
Comment (noiarek) [inline cadence/mapping/grpc/domain.py:147]: Question

File: cadence/mapping/grpc/domain.py [hunk 6]
```diff
@@ -133,13 +139,15 @@ def register_domain_request_dataclass_to_proto(register_domain: RegisterDomainRe
...
         history_archival_status=archival_status_dataclass_to_proto(register_domain.archival_status),
         history_archival_uri=register_domain.archival_bucket_name,
+        # Default value is 0 (invalid), it doesn't accept nil
```
Comment (noiarek) [inline cadence/mapping/grpc/domain.py:150]: Question

File: cadence/mapping/grpc/domain.py [hunk 7]
```diff
@@ -20,69 +26,67 @@ def list_domains_request_dataclass_to_proto(list_domains: ListDomainsRequest) ->
...
-        description=describe_domain.description,
-        owner_email=describe_domain.owner_email,
-        data={key: value for key, value in describe_domain.data.values()},
-        uuid=describe_domain.id,
-    ) if describe_domain else None
+        name=domain.name,
+        status=proto_domain_status_to_dataclass(domain.status),
+        description=domain.description,
+        owner_email=domain.owner_email,
+        data={key: value for key, value in domain.data.items()},
```
Comment (noiarek) [inline cadence/mapping/grpc/domain.py:50]: replacing `values()` to `items()`

**Review by longquanzheng — APPROVED** [review]

**Pull request merged**